### ISSUE TITLE
Scheduled run fails

### ISSUE THREAD

**Issue by ben-manes** [ossf/scorecard-action#84]
I set up this action and the scheduled run [failed](https://github.com/ben-manes/caffeine/runs/5025593098?check_suite_focus=true) with,
```
refs/heads/master not supported with 'schedule' event.
Only the default branch 'refs/heads/null' is supported
```

Since I followed your guide and did not modify this action, I think there is a missing step somewhere?

**Comment by godofredoc** [comment]
Here is an [example](https://github.com/flutter/flutter/runs/5032464875?check_suite_focus=true) from flutter failing because of the same reason.

**Comment by laurentsimon** [comment]
We fixed this in v1.0.3 https://github.com/ossf/scorecard-action/releases/tag/v1.0.3
If you've used the example workflow in the repo or the one from the GitHub UX tab, you're probably using an older version.

In the meanwhile, can you please use `ossf/scorecard-action@ b614d455ee90608b5e36e3299cd50d457eb37d5f `, which is the hash for v1.0.3

Please confirm this works.

Let me update the documentation now.

**Comment by ben-manes** [comment]
In my case I did use `v1.0.3` via a dependabot upgrade. Did the yaml file change between versions otherwise?

**Comment by laurentsimon** [comment]
correct. I was checking your hash and you have the right one. Flutter has the old one. 
Could you run these lines https://github.com/ossf/scorecard-action/blob/main/entrypoint.sh#L51-L54 and tell me what you get?

**Comment by laurentsimon** [comment]
I found the issue. `Repository: null`. I don't understand why the env variable `GITHUB_REPOSITORY` (https://docs.github.com/en/actions/learn-github-actions/environment-variables) is not set for `schedule` trigger, that's strange. Let me see what I can do.

**Comment by ben-manes** [comment]
I created a new token using the same permissions you have in the readme, then ran it:

```sh
curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/ben-manes/caffeine > repo_info.json
ben: caffeine $ echo $SCORECARD_PRIVATE_REPOSITORY
false
ben: caffeine $ echo $SCORECARD_DEFAULT_BRANCH
refs/heads/master

**Comment by laurentsimon** [comment]
We used to take this variable from the json payload https://github.com/ossf/scorecard-action/pull/83/files, but it turns out schedule events don't have one, so I switched to the env variable `GITHUB_REPOSITORY`.. which seems empty as well.
Bear with me. I need to find a way to get the original repository name...

**Comment by ben-manes** [comment]
no worries, thank you!

**Comment by laurentsimon** [comment]
cc @ChrisCarini

**Comment by ScottBrenner** [comment]
Found this issue via a similar failure in https://github.com/ScottBrenner/cfn-lint-action/runs/5074749835?check_suite_focus=true#step:4:15.

> why the env variable GITHUB_REPOSITORY (https://docs.github.com/en/actions/learn-github-actions/environment-variables) is not set for schedule trigger

From https://github.community/t/repository-name-in-environment-variable/16856/7:
> The schedule event does not have an event payload

Reference https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

**Comment by laurentsimon** [comment]
Thank you. I expected the env variable to be set even though there's no even payload.. after all `github.repository` has nothing to do with the even payload, does it?

Thank you all for your help!

**Comment by laurentsimon** [comment]
cc @[GuillaumeRoss](https://github.com/GuillaumeRoss) who reported the same on the scorecard repo https://github.com/ossf/scorecard/issues/1610

**Comment by mjpieters** [comment]
The repository info is available from the git checkout. Bash code I have running elsewhere (but should work on a GH runner after the `actions/checkout` step completed):

```sh
remote=$(git for-each-ref --format='%(upstream:remotename)' "$(git symbolic-ref -q HEAD)")
remote_url=$(git remote get-url "${remote}")
```

This retrieves the remote name for the current HEAD, then gets the URL for that remote. On `actions/checkout` clone that's exactly what you'd want; e.g. from [my failed run logs](https://github.com/mjpieters/SO-userscripts/runs/5135562182?check_suite_focus=true#step:3:30) you can see:

```
  /usr/bin/git remote add origin https://github.com/mjpieters/SO-userscripts
```

and the above bash code will set `remote_url` to `https://github.com/mjpieters/SO-userscripts`.

**Comment by mjpieters** [comment]
Other alternatives:

- Use [`@actions/github`](https://github.com/actions/toolkit/tree/main/packages/github) and use `github.context.repo.owner` plus `github.context.repo.repo`, just like [the `@actions/checkout` code does](https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/input-helper.ts#L20-L24), an combine that with `GITHUB_SERVER_URL`, `GITHUB_URL` or just `https://github.com`, again [same as the checkout action](https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/url-helper.ts#L22-L29).

- The `@actions/github` module gets the repository info from the [Webhook payload](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-common-properties), so find `GITHUB_EVENT_PATH`, load the JSON from that path and you'll have the same data you now get from the `https://api.github.com/repos/$GITHUB_REPOSITORY` api call under the `repository` object.

**Comment by laurentsimon** [comment]
`github.repository` is the backup plan, yes: add an argument and default to `github.repository`. Thanks for confirming. I'd like to avoid creating an additional argument if `GITHUB_REPOSITORY` can be populated by GH automatically - see https://github.com/ossf/scorecard-action/issues/84#issuecomment-1031667855

They are investing atm. The variable should be set, should it not?

**Comment by laurentsimon** [comment]
Finally found some time to dig into this. `GITHUB_REPOSITORY` is not null. I think if you use our repo at head, the schedule works. I've sent an additional https://github.com/ossf/scorecard-action/pull/98 to fix wrong "Is fork" debugging message.

We'll cut a new release as per https://github.com/ossf/scorecard-action/issues/97

**Comment by mjpieters** [comment]
> I think if you use our repo at head, the schedule works.

The schedule does run on repo head, and failed. [This failed, scheduled run](https://github.com/mjpieters/SO-userscripts/runs/5135562182) was executed against the main branch of the repository. The [`@action/checkout` configuration](https://github.com/mjpieters/SO-userscripts/runs/5135562182?check_suite_focus=true#step:3:2-10), with all the action defaults provided by GH:

```
  with:
    persist-credentials: false
    repository: mjpieters/SO-userscripts
    token: ***
    ssh-strict: true
    clean: true
    fetch-depth: 1
    lfs: false
    submodules: false
 ```

checked out [revision c3faf844845191f9a19cf68f1082fb4632e45a9b](https://github.com/mjpieters/SO-userscripts/runs/5135562182?check_suite_focus=true#step:3:118), which was the head at the time of the run:

```
   * [new ref]         c3faf844845191f9a19cf68f1082fb4632e45a9b -> origin/main
```

The `ossf/@scorecard-action` output [agrees it is `refs/heads/main`](https://github.com/mjpieters/SO-userscripts/runs/5135562182?check_suite_focus=true#step:4:10) they are working with, but then bails because:

```
Only the default branch 'refs/heads/null' is supported
```

`null` means the `curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/$GITHUB_REPOSITORY > repo_info.json` fetch failed.

**Comment by mjpieters** [comment]
Side note: I strongly recommend you run `curl` commands with `--fail` or `--fail-with-body` (the latter is a fairly recent option, curl 7.76.0 or newer required) and checking the exit code:

```
if [ ! "$(curl --fail-with-body -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/$GITHUB_REPOSITORY > repo_info.json)" ]; then
    error=$(jq '.message' repo_info.json 2>/dev/null || 'unknown error')
    echo "Failed to get repository information from GitHub: ${error}"
    exit 1;
fi

**Comment by mjpieters** [comment]
> @laurentsimon meant that he thinks this issue was fixed in an unreleased version (probably [6cc0399](https://github.com/ossf/scorecard-action/commit/6cc0399c1f2e07049630e46439784c01e97712da)), so changing the action's hash from `ossf/scorecard-action@b614d45` to `ossf/scorecard-action@869bc60` in your workflow file might resolve the issue.

Ah! Check! That's the part I missed then.

**Comment by laurentsimon** [comment]
Thanks @ben-manes for the clarification, that's indeed what I meant.
@mjpieters thanks for the suggestion about curl's best practice.
